### PR TITLE
Always print explanation of single error message

### LINKED ISSUE #2441: Refactor to always print detailed `--explain` text if there is only a single compiler message

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #2441

I'm not sure if there's something in LSP that needs to be updated as well.

May have merge conflict with #2699

### CODE REVIEW

**Comment by jaschdoc** [comment]
@magnus-madsen ready for review

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 1]
```diff
@@ -224,12 +225,19 @@ object Main {
...
       case Validation.Failure(errors) =>
-        errors.sortBy(_.source.name).foreach(e => println(e.message(flix.getFormatter)))
+        createPrintableMessage(errors.sortBy(_.source.name), flix.getFormatter)
+          .foreach(println)
         println()
         println(s"Compilation failed with ${errors.length} error(s).")
         System.exit(1)
     }
 
+    def createPrintableMessage(errors: LazyList[CompilationMessage], formatter: Formatter): LazyList[String] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:235]: - Add comment.
- Return List
- Rename to `mkMessages`.

(same for the other case)

File: main/src/ca/uwaterloo/flix/util/Formatter.scala [hunk 2]
```diff
@@ -32,7 +32,7 @@ trait Formatter {
     def leftline: String = {
       val sb = new StringBuilder()
       for (lineNo <- beginLine to endLine) {
-        val currentLine = lineAt(lineNo)
+        val currentLine = lineAt
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/util/Formatter.scala:35]: This looks sus

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/util/Formatter.scala]: Should be fixed now

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 3]
```diff
@@ -224,12 +225,23 @@ object Main {
...
         println()
         println(s"Compilation failed with ${errors.length} error(s).")
         System.exit(1)
     }
 
+    /**
+      * Converts a list of compiler error messages to a list of printable messages.
+      * Decides whether or not to print the explanation.
+      */
+    def mkMessage(errors: LazyList[CompilationMessage], formatter: Formatter): List[String] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:239]: This is the same `mkMessage`. What about putting it somewhere? Maybe in the Flix object?

**Comment by magnus-madsen** [comment]
Looks good. But let us see when it is merged in 🤣

**Pull request merged**